components/eamxx/src/physics/spa/spa_functions.hpp

```diff
@@ -124,6 +127,33 @@ struct SPAFunctions
     SPAData         data;         // All spa fields
   }; // SPAInput
 
+  struct IOPReader {
+    IOPReader (iop_ptr_type& iop_,
+               const std::string file_name_,
+               const std::vector<Field>& io_fields_,
+               const std::shared_ptr<const AbstractGrid>& io_grid_)
+      : iop(iop_), file_name(file_name_)
+    {
+      field_mgr = std::make_shared<FieldManager>(io_grid_);
+      for (auto& f : io_fields_) {
+        field_mgr->add_field(f);
+        field_names.push_back(f.name());
+      }
+
+      // Set IO info for this grid and file in IOP object
+      iop->setup_io_info(file_name, io_grid_);
+    }
+
+    void read_variables(const int time_index) {
+      iop->read_fields_from_file_for_iop(file_name, field_names, util::TimeStamp(), field_mgr, time_index);
+    }
+
+    iop_ptr_type iop;
+    std::string file_name;
+    std::vector<std::string> field_names;
+    std::shared_ptr<FieldManager> field_mgr;
+  };
```

Comment:
Reader struct for IOP. Analogous to the `AtmosphereInput` class where we give a list of fields to read, and define a `read_variables()` function.